src/components/main/friends/mod.rs

```diff
@@ -39,41 +30,15 @@ pub struct Props {
 pub fn Friends(cx: Scope<Props>) -> Element {
     log::debug!("rendering Friends");
     let add_error = use_state(&cx, String::new);
-    let disp_friends = use_state(&cx, Vec::new);
-    let friends = use_ref(&cx, HashSet::new);
 
-    let state = use_atom_ref(&cx, STATE).clone();
-    let sidebar_visibility = match state.read().hide_sidebar {
+    let show_friend_list = use_state(&cx, || true);
+
+    let st = use_atom_ref(&cx, STATE).clone();
```

Comment:
There is no need to clone here (though looking at the commit history it looks like its just being carried over so this isnt a blocker but purely a suggestion). Doing this as suggested is best then if you need a copy of it for whatever reason to clone. 
```suggestion
    let state = use_atom_ref(&cx, STATE);
    let st = state.clone();
```

Reply:
Good point, I just deleted that clone and renamed it.